python/ray/data/_internal/plan.py

```diff
@@ -568,20 +568,14 @@ def execute(
                 )
         if not self.has_computed_output():
             if self._run_with_new_execution_backend():
-                from ray.data._internal.execution.bulk_executor import BulkExecutor
                 from ray.data._internal.execution.legacy_compat import (
                     execute_to_legacy_block_list,
                 )
                 from ray.data._internal.execution.streaming_executor import (
                     StreamingExecutor,
                 )
 
-                if context.use_streaming_executor:
```

Comment:
@scottjlee anything preventing us from removing `use_streaming_executor` from `DataContext`?

Reply:
i'm not sure, but i think we have several related fields like enable_optimizer as well, that we discussed and decided we want to remove together in a separate PR